src/inmanta/server/services/orchestrationservice.py

```diff
@@ -673,12 +673,15 @@ async def _check_version_info(
         )
         for inmanta_module_name, module_data in module_version_info.items():
             module_version = module_data.version
+            if not module_data.for_agents:
+                # No agent is using this module in this version
+                continue
```

Comment:
@sanderr Do you agree this was a bug ? It triggers e.g. in the following lsm test case:
I think the following is happening:
- The `unittest` module is not ever used but:
        - it gets registered during the exports -> will be returned by `code_manager.get_module_version_info()` with en empty `for_agents=[]`
        - it will never be registered in the `AgentModules` table.
        
Is this correct ? i.e. only checking version for modules which are actually used in the current version ?
 https://github.com/inmanta/lsm/blob/8ec018f10c3717c07d38c156d2d39ea8613defef/tests/integration/test_service_entities_exporter.py#L3518

Reply:
I agree there's a bug here. But I'm not sure this is an acceptable fix. You're checking the current export's `for_agents` (right?). Meaning that you only check that it's not used by the new resources, but it may still be used by existing resources from the base version.

Either we need to be smarter about it, or we need to make sure it gets registered in the table anyway. I need to give this more thought.

Reply:
>  You're checking the current export's for_agents (right?)  
Yes